web/components/shared/themed/themedTableHeader.tsx

```diff
@@ -79,12 +87,33 @@ const notificationMethods = [
 ];
 
 export default function ThemedHeader(props: ThemedHeaderProps) {
-  const { isFetching, editColumns, timeFilter, advancedFilter, csvExport } =
-    props;
+  const {
+    isFetching,
+    editColumns,
+    timeFilter,
+    advancedFilter,
+    csvExport,
+    filterLayouts,
+  } = props;
 
   const [showAdvancedFilters, setShowAdvancedFilters] = useState(false);
   const [exportFiltered, setExportFiltered] = useState(false);
 
+  const [isSaveFiltersModalOpen, setIsSaveFiltersModalOpen] =
+    useState<boolean>(false);
+
+  const [filterName, setFilterName] = useState<string>("");
+
+  const onSaveFilter = () => {
+    filterLayouts?.onSaveFilter?.(filterName) ?? (() => {})();
```

Comment:
why do we have the `(() => {})();` after? looks kinda funny. 

Figure it'd be something like this: 
`if (filterLayouts?.onSaveFilter) { ... }`